lib/ansible/module_utils/facts/network/linux.py

```diff
@@ -147,9 +147,13 @@ def get_interfaces_info(self, ip_path, default_ipv4, default_ipv6):
                 interfaces[device]['perm_macaddress'] = get_file_content(os.path.join(path, 'bonding_slave', 'perm_hwaddr'), default='')
             if os.path.exists(os.path.join(path, 'device')):
                 interfaces[device]['pciid'] = os.path.basename(os.readlink(os.path.join(path, 'device')))
+            if os.path.exists(os.path.join(path, 'duplex')):
+                duplex = get_file_content(os.path.join(path, 'duplex'))
+                if duplex is not None and duplex != 'unknown':
```

Comment:
``` python
     duplex = get_file_content(os.path.join(path, 'duplex'))
     if duplex is not None:
         interfaces[device]['duplex'] = duplex     
```

get_file_content() will do a os.path.exists(), so no need to do it before the call to get_file_content